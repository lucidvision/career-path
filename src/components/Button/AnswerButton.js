import React from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, Text, TouchableOpacity } from 'react-native'

AnswerButton.propTypes = {
  onPress: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired
}

AnswerButton.defaultProps = {
  onPress: () => {},
  title: ''
}

export default function AnswerButton (props) {
  const { style, title, ...rest } = props
  return (
    <TouchableOpacity style={[styles.button, style]} {...rest}>
      <Text style={styles.buttonText}>{title}</Text>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    backgroundColor: 'blue',
    borderRadius: 30,
    height: 80,
    justifyContent: 'center',
    padding: 10
  },
  buttonText: {
    color: 'white',
    fontSize: 20
  }
})
