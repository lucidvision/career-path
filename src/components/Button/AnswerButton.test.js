import React from 'react'
import { shallow } from 'enzyme'
import AnswerButton from './AnswerButton'

const createAnswerButton = (override) => {
  const props = {
    onPress: () => {},
    style: {},
    title: '',
    ...override
  }

  return shallow(<AnswerButton {...props} />)
}

describe('AnswerButton', () => {
  const component = createAnswerButton()

  it('renders correctly', () => {
    expect(component).toMatchSnapshot()
  })

  describe('renders the correct output', () => {
    it('displays the correct text', () => {
      const component = createAnswerButton({title: 'Rainbow'})
      expect(component.find('Text').props().children).toBe('Rainbow')
    })

    it('triggers callback on press', () => {
      const mockOnPress = jest.fn()
      const component = createAnswerButton({onPress: mockOnPress})
      const touchableOpacity = component.find('TouchableOpacity')
      touchableOpacity.props().onPress()
      expect(mockOnPress).toHaveBeenCalled()
    })
  })
})
