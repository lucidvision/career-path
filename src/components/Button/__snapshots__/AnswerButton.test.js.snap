// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerButton renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "blue",
        "borderRadius": 30,
        "height": 80,
        "justifyContent": "center",
        "padding": 10,
      },
      Object {},
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 20,
      }
    }
  />
</TouchableOpacity>
`;
